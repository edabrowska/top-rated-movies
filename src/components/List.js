import React, { useEffect } from 'react'
import { connect } from 'react-redux'

import { getMovies } from '../store/actions/index'

const ConnectedList = ({ movies, getMovies }) => {

  useEffect(() => {
    getMovies()
  }, [])

console.log(movies)
  return (
  <ul className='list'>
    {movies && movies.map(el => (
      <li key={el.id}>{el.title} {el.release_date} {el.vote_average}</li>
    ))}
  </ul>
)}

const List = connect(
  state => ({
    movies: state.movies.slice(0, 20)
  }),
  { getMovies }
)(ConnectedList)

export default List
