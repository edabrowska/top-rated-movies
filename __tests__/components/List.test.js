import React from 'react'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'

import initialState from '../../__mocks__/fileMock'

import List from '../../src/components/List'

describe('List', () => {
  const movies = initialState.movies
  const mockStore = configureStore()

  it('Snapshot', () => {
    const component = renderer.create(
      <Provider store={mockStore(initialState)}>
        <List movies={movies} />
      </Provider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
