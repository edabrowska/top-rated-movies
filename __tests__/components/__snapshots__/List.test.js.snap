// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Snapshot 1`] = `
<ul
  className="list"
>
  <li
    className="movie-tile"
  >
    <div
      className="movie-tile__wrapper-left"
    >
      <h3
        className="movie-tile__title"
      >
        Ad Astra
      </h3>
      <p>
        2019-09-17
      </p>
    </div>
    <div
      className="movie-tile__wrapper-right"
    >
      <p
        className="movie-tile__rating"
      >
        6
      </p>
      <button
        className="movie-tile__button"
        onClick={[Function]}
      >
        INFO
      </button>
    </div>
  </li>
  <li
    className="movie-tile"
  >
    <div
      className="movie-tile__wrapper-left"
    >
      <h3
        className="movie-tile__title"
      >
        Birds of Prey (and the Fantabulous Emancipation of One Harley Quinn)
      </h3>
      <p>
        2020-02-05
      </p>
    </div>
    <div
      className="movie-tile__wrapper-right"
    >
      <p
        className="movie-tile__rating"
      >
        7.1
      </p>
      <button
        className="movie-tile__button"
        onClick={[Function]}
      >
        INFO
      </button>
    </div>
  </li>
</ul>
`;
